sql-plugin/src/main/spark330db/scala/org/apache/spark/sql/rapids/execution/GpuExecutorBroadcastHelper.scala

```diff
@@ -65,11 +66,31 @@ object GpuExecutorBroadcastHelper {
     // Use the GPU Shuffle Coalesce iterator to concatenate and load batches onto the 
     // host as needed. Since we don't have GpuShuffleCoalesceExec in the plan for the 
     // executor broadcast scenario, we have to use that logic here to efficiently 
-    // grab and release the semaphore while doing I/O
+    // grab and release the semaphore while doing I/O. We wrap this with GpuCoalesceIterator
+    // to ensure this always a single batch for the following step.
+    val timeMetrics = Set(CONCAT_TIME, OP_TIME)
+    val shuffleMetrics = metricsMap.map { case (k, v) =>
+      if (timeMetrics.contains(k)) {
+        (k, v)
+      } else {
+        (k, NoopMetric)
+      }
+    }
```

Comment:
Nit: We're iterating the larger map rather than the smaller one, and we don't need to iterate at all.
```suggestion
    val shuffleMetrics = Map(
      CONCAT_TIME -> metricsMap(CONCAT_TIME),
      OP_TIME -> metricsMap(OP_TIME))
```

Reply:
So we do it this way because GpuShuffleCoalesceIterator and HostShuffleCoalesceIterator expect many of those metrics to be in the metricsMap, so we're just replacing these effectively with a default value. I think I can do a cleaner version using withDefaultValue.

Reply:
Ah, good catch, I meant to put `withDefaultValue` in my example.